src/components/frontend-engine/yup/helper.ts

```diff
@@ -57,7 +57,7 @@ export namespace YupHelper {
 						Object.keys(parsedRule.when).forEach((whenFieldId) => {
 							parsedRule.when[whenFieldId] = {
 								...parsedRule.when[whenFieldId],
-								yupSchema: parsedFieldConfigs[whenFieldId].schema,
+								yupSchema: parsedFieldConfigs[whenFieldId]?.schema,
```

Comment:
hmm then the rule does not have info on the base schema of the dependent field? would it break validation?

Reply:
yeah it may break, unfortunately there is no good way to know when the validation schema is "final" so the onus will be on the developer to ensure the conditional validation works correctly